arviz/plots/distplot.py

```diff
@@ -142,6 +144,16 @@ def plot_dist(
 
         >>> az.plot_dist(b, rug=True, quantiles=[.25, .5, .75], cumulative=True)
     """
+    if isinstance(values, xr.Dataset):
+        raise ValueError(
+            "Cannot directly convert xarray.Dataset to numpy array."
+            " Instead, create an xarray.DataArray first "
+            "or Use plot_posterior, plot_density, plot_joint"
+            "or plot_pair instead of plot_dist"
+        )
+    if isinstance(values, InferenceData):
+        raise ValueError(" Inference Data object detected. Use plot_posterior instead of plot_dist")
```

Comment:
Even though everything it says is correct, I would change it slightly to guide users towards our intended workflow, moreover, I figured we can have a single message for both cases:

```
    if isinstance(values, xr.Dataset) or isinstance(values, InferenceData):
        raise ValueError(
            "InferenceData or xarray.Dateset object detected,"
            " use plot_posterior, plot_density or plot_pair"
            " instead of plot_dist"
        )
```

They can always create a DataArray manually, but we have the `var_names` argument so users _should not_ create dataarrays manually. We also recently deprecated `plot_joint` in favour of `plot_pair` so error messages should not continue recommending its usage.